drivers/display/display_nrf_led_matrix.c

```diff
@@ -15,25 +15,48 @@
 #ifdef PPI_PRESENT
 #include <nrfx_ppi.h>
 #endif
+#include <nrf_peripherals.h>
+#include <logging/log.h>
+LOG_MODULE_REGISTER(nrf_led_matrix, CONFIG_DISPLAY_LOG_LEVEL);
 
 #define MATRIX_NODE  DT_INST(0, nordic_nrf_led_matrix)
 #define TIMER_NODE   DT_PHANDLE(MATRIX_NODE, timer)
 #define USE_PWM      DT_NODE_HAS_PROP(MATRIX_NODE, pwm)
+#define PWM_NODE     DT_PHANDLE(MATRIX_NODE, pwm)
 #define ROW_COUNT    DT_PROP_LEN(MATRIX_NODE, row_gpios)
 #define COL_COUNT    DT_PROP_LEN(MATRIX_NODE, col_gpios)
+#define GROUP_SIZE   DT_PROP(MATRIX_NODE, pixel_group_size)
+#if (GROUP_SIZE > DT_PROP(TIMER_NODE, cc_num) - 1) || \
+    (USE_PWM && GROUP_SIZE > PWM0_CH_NUM)
+#error "Invalid pixel-group-size configured."
+#endif
```

Comment:
nit: better BUILD_ASSERT?

Reply:
With `#error`, the actual error message is more visible in the output:
```
/home/andy/zephyrproject/zephyr/drivers/display/display_nrf_led_matrix.c:38:2: error: #error "Invalid length of pixel-mapping."
   38 | #error "Invalid length of pixel-mapping."
      |  ^~~~~
ninja: build stopped: subcommand failed.
```
vs.
```
/home/andy/zephyrproject/zephyr/include/toolchain/gcc.h:61:36: error: static assertion failed: "Invalid length of pixel-mapping."
   61 | #define BUILD_ASSERT(EXPR, MSG...) _Static_assert(EXPR, "" MSG)
      |                                    ^~~~~~~~~~~~~~
/home/andy/zephyrproject/zephyr/drivers/display/display_nrf_led_matrix.c:37:1: note: in expansion of macro 'BUILD_ASSERT'
   37 | BUILD_ASSERT(PIXEL_COUNT == (X_PIXELS * Y_PIXELS),
      | ^~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```
That's why I used `#error`s where it was possible. Are there some advantages of build assertions in those cases?

Reply:
In this case, not really, preprocessor can handle these expressions. I think `BUILD_ASSERT` can handle stuff like `sizeof`, so it should be able to cover more use cases.